core/xpdo/xpdo.class.php

```diff
@@ -2076,8 +2076,10 @@ protected function _log($level, $msg, $target= '', $def= '', $file= '', $line= '
         $content= @ob_get_contents();
         @ob_end_clean();
         if ($target=='FILE' && $this->getCacheManager()) {
-            $filename = isset($targetOptions['filename']) ? $targetOptions['filename'] : 'error.log';
-            $filepath = isset($targetOptions['filepath']) ? $targetOptions['filepath'] : $this->getCachePath() . xPDOCacheManager::LOG_DIR;
+            $defaultFilename = !empty($this->getOption('error_log_filename')) ? $this->getOption('error_log_filename') : 'error.log';
```

Comment:
This is failing on PHP 5.4 with error:
> Fatal error: Can't use method return value in write context in /home/travis/build/modxcms/revolution/core/xpdo/xpdo.class.php on line 2079


The problem is the `empty` method being called on the result of `$this->getOption()`. Old PHP doesn't seem to like that.

Instead of using these ternary operators, you can use the following:

`$defaultFilename = $this-getOption('error_log_filename', null, 'error.log', true);`

The last `true` indicates an empty() check will be run on the value. Looks a bit cleaner, too. 

This comment applies to the other files, too.